src/gc/gc.d

```diff
@@ -1671,6 +1679,43 @@ struct Gcx
         return pool.npages * PAGESIZE;
     }
 
+    /**
+     * Update the thresholds for when to collect the next time
+     */
+    void updateCollectThresholds() nothrow
+    {
+        static float max(float a, float b) nothrow
+        {
+            return a >= b ? a : b;
+        }
+
+        static size_t roundUp(float val, size_t multiple) nothrow
+        {
+            return cast(size_t)(val + multiple - 1 + 0.5) / multiple * multiple;
+        }
+
+        static float zeroDenormal(float val) nothrow
+        {
+            return val <= float.min_normal ? 0 : val;
+        }
+
+        static float decay(float oldVal, float newVal) nothrow
+        {
+            // one-pole RC low-pass
+            // see http://en.wikipedia.org/wiki/RC_time_constant
+            // and http://en.wikipedia.org/wiki/Low-pass_filter#Simple_infinite_impulse_response_filter
```

Comment:
Tried to simplify it a little, but a reference to how the magical constants were computed still makes sense IMO.